src/mgmt/cli_mgmt.c

```diff
@@ -873,21 +872,32 @@ attach(const char *name, struct dc_mgmt_sys **sysp)
 		goto out;
 	}
 
-	rc = get_attach_info(name, true /* all_ranks */, &sys->sy_info, &resp);
-	if (rc != 0)
-		goto err_sys;
+	if (!resp_g) {
+		need_free_resp = true;
+		rc = get_attach_info(name, true /* all_ranks */, &sys->sy_info, &resp);
+		if (rc != 0)
+			goto err_sys;
+	} else {
+		resp = resp_g;
```

Comment:
Do we need to check if `name` is the same with `agent_sys_name`? Not sure if my understanding of `agent_sys_name` is correct.

Reply:
the problem is that today we only support 1 system per agent. so unless the agent can support multiple systems then this check is not necessary.

Reply:
In theory a host could have multiple agents, though, and a client application may want access to both. IMO if a client has requested a system whose info isn't cached, it's worth it to try fetching it, and see if the name has changed.